src/client/src/Hooks/useTasks.js

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from "react";
+import { GetResourceRequest } from "../Api/ApiManger";
+
+export const useTasks = (rerender) => {
+    const [tasks, setTasks] = useState([]);
+
+    useEffect(() => {
+        const getTask = async () => {
```

Comment:
You are missing a condition `if (rerender)`
Without this condition you will rerender everytime the value of rerender is changing